### PR TITLE
Photo long list bug perf

### PR DESCRIPTION
**Pull request by nono**
I've taken the work made by @Benibur and fixed small stuff.

### CODE REVIEW

**Comment by frankrousseau** [comment]
OK to take the @Benibur work. But it requires it follows the guidelines. Can you lint everything?

**Comment by nono** [comment]
@frankrousseau can you be more precise? `npm run lint` is green. It checks only for the server-side. Most lint errors for the client are related to the maximum line length and were there before this PR.

File: client/app/views/file_view.coffee [hunk 1]
```diff
@@ -383,29 +430,44 @@ module.exports = class FileView extends BaseView
...
-            return @$(selector)?[0] or null
+        if @$el.hasClass('edit-mode')
+            return
 
-        # For each forbidden element, check if it, or one of its children, has
-        # been clicked.
-        results = forbiddenElements.filter (element) ->
-            return element? and
-                (element is event.target or $.contains(element, event.target))
+        t = event.target
```
Comment (frankrousseau) [inline client/app/views/file_view.coffee]: Here is a single letter variable. It doesn't follow our guidelines.

File: server/controllers/files.coffee [hunk 2]
```diff
@@ -458,3 +449,55 @@ module.exports.search = (req, res, next) ->
...
+
+###*
+ * Returns "screens" (image reduced ) for a given file.
+ * There is a bug : when the browser cancels many downloads, some are not
+ * cancelled, what leads to saturate the stack of threads and blocks the
+ * download of thumbs.
+ * Cf comments bellow to reproduce easily
+###
+module.exports.photoScreen = (req, res, next) ->
+    if !res.connection or res.connection.destroyed
```
Comment (frankrousseau) [inline server/controllers/files.coffee]: Here it uses ! instead of not

File: client/app/router.coffee [hunk 3]
```diff
@@ -50,6 +50,13 @@ module.exports = class Router extends Backbone.Router
         app.baseCollection.getByFolder folderID, (err, folder, collection) =>
             if err? then console.log err
             else
+                # sort by creation date the folder where phones photo are
+                # uploaded
+                path = folder.getRepository()
+                if  path == '/Appareils photo' or path == '/Photos from devices'
```
Comment (frankrousseau) [inline client/app/router.coffee]: Here is a == instead of is

File: client/app/views/file_info.coffee [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        elmt.addEventListener 'mouseenter', (event) =>
+            @_isIntoPopover = true
+            @stateMachine.E3_enterPopo()
+        elmt.addEventListener 'mouseleave', (event) =>
+            @_isIntoPopover = false
+            @stateMachine.E4_exitPopo()
+        @a.addEventListener 'click', (event) =>
+            # if ctrl => open link in new window : nothing to do
+            if event.ctrlKey
+                return
```
Comment (frankrousseau) [inline client/app/views/file_info.coffee]: Here is return instead of a if ... else.

**Comment by frankrousseau** [comment]
I added comments on some commits to show you what I meant.

**Comment by nono** [comment]
@frankrousseau OK, it's done

**Comment by frankrousseau** [comment]
Great, thank you!

**Pull request merged**